src/XrdCl/XrdClXCpSrc.cc

```diff
@@ -94,7 +94,7 @@ class ChunkHandler: public ResponseHandler
       {
         ChunkInfo *rsp = nullptr;
         response->Get( rsp );
-        *chunk = PageInfo( rsp->offset, rsp->length, rsp->buffer );
+        chunk = new PageInfo( rsp->offset, rsp->length, rsp->buffer );
         response->Set( ( int* )0 );
```

Comment:
You are absolutely correct that the `PageInfo` object needs to be allocated. Just to make the patch complete, could you also remove the line 98 as the response (`AnyObject`) needs to maintain the ownership of the `ChunkInfo`?